WFI condition fix

ctopal opened the pull request:
Before this commit Spike was requiring S mode privilege even without
S mode implemented. This commit fixes it.

Signed-off-by: Canberk Topal <ctopal@lowrisc.org>

GregAC reviewed (approved):
LGTM, worth submitting upstream also

GregAC commented:
Looking at it again the initial code may have been correct. WFI is valid in U mode if the implementation supports U mode interrupts. Ibex does not, where Spike might.

The bit you've changed looks like it relates to virtualisation so hard to comment on whether the change is correct, instead the better change may be:

``` c
} else if (STATE.v) { // VS-mode
  if (get_field(STATE.hstatus->read(), HSTATUS_VTW))
    require_novirt();
  else
    require_privilege(PRV_S);
} else {
  require_privilege(PRV_S);
}
wfi();
```

ctopal commented:
Okay I think I know where the require privilege S comes from. In the spec it says: 

```
When S-mode is implemented, then executing WFI in U-mode causes an illegal instruction exception.
```

However, in Ibex we don't implement S-mode which then allows us to execute WFI in U-mode freely!

ctopal committed:
WFI condition fix

Before this commit Spike was requiring S mode privilege even without
S mode implemented. This commit fixes it.

Signed-off-by: Canberk Topal <ctopal@lowrisc.org>

ctopal commented:
@GregAC Should I merge this version? I tested it with our environment, seems to be working and in Machine Level ISA v1.12, it says:
```
When S-mode is implemented, then executing WFI in U-mode causes an illegal instruction exception, unless it completes within an implementation-specific, bounded time limit. 
```
Which justifies the check for S-mode implementation before throwing illegal instruction exception. WDYT?

ctopal commented:
Got merged in upstream, doing the same here.
